### PR TITLE
Generate report

### LINKED ISSUE #128: Create basic pdf summary
## Description
write code to generate a basic pdf summary of this weeks data.

## Required Files
- `generate_report.py`

## User Story
- As an engineer it is important to be able to provide multiple ways for the users to access the information required.

### PR DESCRIPTION
**Pull request by shindym**
- Generates basic report: Summary and individual websites
- Diagram folder added
- .gitignore updated

Closes #128

### CODE REVIEW

File: weekly_report/generate_report.py [hunk 1]
```diff
@@ -1,3 +1,499 @@
...
+from dotenv import load_dotenv
+from reportlab.lib.pagesizes import letter
+from reportlab.pdfgen import canvas
+from reportlab.pdfbase import pdfmetrics
+from reportlab.pdfbase.ttfonts import TTFont
+from psycopg2 import connect
+from psycopg2.extras import RealDictCursor
+from psycopg2.extensions import connection
+import altair as alt
+import pandas as pd
```
Comment (annalisev) [inline weekly_report/generate_report.py:11]: can we split these up at all?

File: weekly_report/generate_report.py [hunk 2]
```diff
@@ -1,3 +1,499 @@
+import os
```
Comment (annalisev) [inline weekly_report/generate_report.py:1]: Docstring here?

File: weekly_report/generate_report.py [hunk 3]
```diff
@@ -1,3 +1,499 @@
...
+import altair as alt
+import pandas as pd
+
+
+class Stats_Retriever():
+    """This class is responsible for the functionality which allows the user
+    to retrieve useful stats about the games database. """
+
+    def __init__(self, config=None, website_id=(1, 2, 3)) -> None:
+        """Initialises DB connection and website ids """
```
Comment (annalisev) [inline weekly_report/generate_report.py:19]:
```suggestion
        """Initialises DB connection and website IDs. """
```

File: weekly_report/generate_report.py [hunk 4]
```diff
@@ -1,3 +1,499 @@
...
+                        AND g.release_date <= CURRENT_DATE AND g.website_id IN %s) GROUP BY t.tag_name ORDER BY count(*) DESC LIMIT 5;
+                        """, (self.website_ids,))
+            tags = cur.fetchall()
+            tag_list = []
+            for x in range(len(tags)):
+                tag_list.append(tags[x]['tag_name'])
+            return tag_list
+
+    def tag_game_ratio(self):
+        """Returns the proportion of tags in the form of a pie-chart. """
```
Comment (annalisev) [inline weekly_report/generate_report.py:139]:
```suggestion
        """Returns the proportion of tags in the form of a pie-chart."""
```

**Review by annalisev — COMMENTED** [review]
Looks good, just a couple of small suggestions!

File: weekly_report/generate_report.py [hunk 5]
```diff
@@ -1,3 +1,499 @@
...
+from psycopg2.extensions import connection
+import altair as alt
+import pandas as pd
+
+
+class Stats_Retriever():
+    """This class is responsible for the functionality which allows the user
+    to retrieve useful stats about the games database. """
+
+    def __init__(self, config=None, website_id=(1, 2, 3)) -> None:
```
Comment (hhm970) [inline weekly_report/generate_report.py:18]: Might be useful to specify a type hint for `website_id`

**Review by hhm970 — COMMENTED** [review]
Well done with your consistency across this script, especially with its size!

As far as I can tell, there are no explicit errors/discrepancies in your code. I think one thing you could do to improve is add type hints in your functions' arguments

File: weekly_report/generate_report.py (hunk 5 above)
Comment (shindym) [reply weekly_report/generate_report.py:18]: Will add!

File: weekly_report/generate_report.py [hunk 6]
```diff
@@ -1,3 +1,499 @@
...
+            cur.execute(
+                """SELECT w.website_name FROM website AS w WHERE w.website_id IN %s""", (self.website_ids,))
+            websites = cur.fetchall()
+            website_list = []
+            for x in range(len(websites)):
+                website_list.append(websites[x]['website_name'])
+            return website_list
+
+
+def generate_pdf(config):
```
[the remark this answers is not in the document]
Comment (shindym) [reply weekly_report/generate_report.py:225]: Unsure but will look into it.

**Review by annalisev — APPROVED** [review]
The changes look good!

File: weekly_report/generate_report.py [hunk 7]
```diff
@@ -1,3 +1,391 @@
...
+from psycopg2 import connect
+from psycopg2.extras import RealDictCursor
+from psycopg2.extensions import connection
+import altair as alt
+import pandas as pd
+
+
+class StatsRetriever():
+    """This class is responsible for the functionality which allows the user
+    to retrieve useful stats about the games database. """
```
Comment (annalisev) [inline weekly_report/generate_report.py:18]:
```suggestion
    to retrieve useful stats about the games database."""
```

File: weekly_report/generate_report.py [hunk 8]
```diff
@@ -1,3 +1,391 @@
...
+import altair as alt
+import pandas as pd
+
+
+class StatsRetriever():
+    """This class is responsible for the functionality which allows the user
+    to retrieve useful stats about the games database."""
+
+    def __init__(self, config=None, website_id: tuple = (1, 2, 3)) -> None:
+        """Initialises DB connection and website IDs. """
```
Comment (annalisev) [inline weekly_report/generate_report.py:21]:
```suggestion
        """Initialises DB connection and website IDs."""
```

**Review by annalisev — APPROVED** [review]
Looks good!

**Pull request merged**